src/handlers/gestures/gesture.ts

```diff
@@ -0,0 +1,289 @@
+  public handlerName = '';
+  public config: BaseGestureConfig = {};
+  public handlers: HandlerCallbacks<EventPayloadT> = {
+    handlerTag: -1,
+  };
+
+  private addDependency(
+    key: 'simultaneousWith' | 'requireToFail',
+    gesture: BaseGesture<unknown> | React.RefObject<BaseGesture<unknown>>
+  ) {
+    this.config[key] = this.config[key]
+      ? [].concat(this.config[key] as never, gesture as never)
+      : [gesture];
+  }
+
+  setRef(ref: React.MutableRefObject<BaseGesture<unknown>>) {
+    this.config.ref = ref;
+    return this;
+  }
+
+  setOnBegan(
+    callback: (
+      event: UnwrappedGestureHandlerStateChangeEvent<EventPayloadT>
+    ) => void
+  ) {
+    this.handlers.onBegan = callback;
+    return this;
+  }
+
+  setOnStart(
+    callback: (
+      event: UnwrappedGestureHandlerStateChangeEvent<EventPayloadT>
+    ) => void
+  ) {
+    this.handlers.onStart = callback;
+    return this;
+  }
+
+  setOnEnd(
+    callback: (
+      event: UnwrappedGestureHandlerStateChangeEvent<EventPayloadT>,
+      success: boolean
+    ) => void
+  ) {
+    this.handlers.onEnd = callback;
+    return this;
+  }
+
+  setEnabled(enabled: boolean) {
+    this.config.enabled = enabled;
+    return this;
+  }
+
+  setMinPointers(minPointers: number) {
+    this.config.minPointers = minPointers;
+    return this;
+  }
+
+  setShouldCancelWhenOutside(value: boolean) {
+    this.config.shouldCancelWhenOutside = value;
+    return this;
+  }
+
+  setHitSlop(hitSlop: HitSlop) {
+    this.config.hitSlop = hitSlop;
+    return this;
+  }
+
+  addSimultaneousGesture(
+    gesture: BaseGesture<any> | React.RefObject<BaseGesture<any>>
+  ) {
+    this.addDependency('simultaneousWith', gesture);
+
+    return this;
+  }
+
+  addRequiredToFailGesture(
+    gesture: BaseGesture<any> | React.RefObject<BaseGesture<any>>
+  ) {
+    this.addDependency('requireToFail', gesture);
+
+    return this;
+  }
+
+  simultaneousWith(other: BaseGesture<any>): InteractionBuilder {
+    return new InteractionBuilder(this).simultaneousWith(other);
+  }
+
+  exclusiveWith(other: BaseGesture<any>): InteractionBuilder {
+    return new InteractionBuilder(this).exclusiveWith(other);
+  }
+
+  requireToFail(other: BaseGesture<any>): InteractionBuilder {
+    return new InteractionBuilder(this).requireToFail(other);
+  }
+
+  initialize() {
+    this.handlerTag = getNextHandlerTag();
+    this.handlers = { ...this.handlers, handlerTag: this.handlerTag };
+
+    if (this.config.ref) {
+      this.config.ref.current = this;
+    }
+  }
+
+  configure(): BaseGesture<any>[] {
+    return [this];
+  }
+
+  prepare() {
+    if (this.config.requireToFail !== undefined) {
+      this.config.requireToFail = this.toArray(this.config.requireToFail);
+    }
+
+    if (this.config.simultaneousWith !== undefined) {
+      this.config.simultaneousWith = this.toArray(this.config.simultaneousWith);
+    }
+  }
+
+  private toArray(x: unknown) {
+    return [].concat(x as never);
```

Comment:
Why we cast to never here?

Reply:
Fixed in 6da30cb388f462568017bb46206988886afd142d